routers/admin/admin.go

```diff
@@ -202,6 +202,17 @@ func SendTestMail(ctx *context.Context) {
 	ctx.Redirect(setting.AppSubURL + "/admin/config")
 }
 
+func shadowPassword(cfgItem string) string {
+	fields := strings.Split(cfgItem, ",")
```

Comment:
I think there need to be more than one check, since the config strings can be different for MySQL and redis. This seems to fix for redis, but not MySQL. In the example from  #7147, the session provider connection string is:
```
someclient:somepassword@tcp(srv-mysql:3306)/someclient
```
This format uses DSN, so the password is optional: https://github.com/go-sql-driver/mysql

Maybe it you can also pass in the adapter/provider to shadowPassword and then know if it is Redis/MySQL and check based on that.

Reply:
OK. Will fix that.